send-emails-from-a-service/README.md

```diff
@@ -0,0 +1,682 @@
+        return <http:Created>{};
+    }
+}
+
+function getEmailContent(int appointmentNumber, Appointment appointment, Payment payment)
+        returns string {
+    return string `Appointment Confirmation
+
+    Appointment Details
+        Appointment Number : ${appointmentNumber}
+        Appointment Date: ${appointment.appointmentDate}
+
+    Patient Details
+        Name : ${appointment.patient.name}
+        Contact Number : ${appointment.patient.phone}
+
+    Doctor Details
+        Name : ${appointment.doctor.name}
+        Specialization : ${appointment.doctor.category}
+
+    Payment Details
+        Doctor Fee : ${payment.actualFee}
+        Discount : ${payment.discount}
+        Total Fee : ${payment.discounted}
+        Payment Status : ${payment.status}`;
+}
+```
+
+- The first backend call is a `POST` request to the hospital service to reserve the appointment. The `hospital_id` and `category` values are used as path parameters.
+
+```ballerina
+Appointment|http:ClientError appointment =
+        hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+    patient,
+    doctor,
+    hospital,
+    appointment_date
+});
+```
+
+Use the `is` check to decide the flow based on the response to the client call. If the request failed, return a "NotFound" response. Else, if the payload could not be bound to `Appointment` as expected or if there were any other failures, respond with an "InternalServerError" response.
+
+```ballerina
+if appointment !is Appointment {
+    log:printError("Appointment reservation failed", appointment);
+    if appointment is http:ClientRequestError {
+        return <http:NotFound>{body: string `unknown hospital, doctor, or category`};
+    }
+    return <http:InternalServerError>{body: appointment.message()};
+}
+```
+
+If the appointment reservation was successful, we can make the payment by making a `POST` request to the payment service. The payload includes details extracted out from the original request (for `card_number`), the appointment reservation response (for `appointmentNumber`, `doctor`, `patient`, `fee` and `confirmed`)
+
+```ballerina
+Payment|http:ClientError payment = paymentEP->/.post({
+    appointmentNumber,
+    doctor: appointment.doctor,
+    patient: appointment.patient,
+    fee: appointment.doctor.fee,
+    confirmed: appointment.confirmed,
+    card_number: cardNo
+});
+
+if payment !is Payment {
+    log:printError("Payment settlement failed", payment);
+    if payment is http:ClientRequestError {
+        return <http:NotFound>{body: string `unknown appointment number`};
+    }
+    return <http:InternalServerError>{body: payment.message()};
+}
+```
+
+If the payment was successful, the next and final step is to send an email to the user containing the appointment details as the email body, namely, `Appointment Details`, `Patient Details`, `Doctor Details` and `Payment Details`. Then we create the email message with the user's email, subject and the email body.
```

Comment:
Bullet point.

Reply:
Do we really need the "namely, `Appointment Details`, `Patient Details`, `Doctor Details` and `Payment Details`" part?

Reply:
Need a comma before " and the email body".

Reply:
> Do we really need the "namely, `Appointment Details`, `Patient Details`, `Doctor Details` and `Payment Details`" part?

Removed that part